sale_product_company/migrations/16.0.2.0.0/pre-migration.py

```diff
@@ -0,0 +1,27 @@
+# Copyright 2024 Tecnativa - Pilar Vargas
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openupgradelib import openupgrade
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    if not openupgrade.table_exists(env.cr, "product_template_res_company_rel"):
+        return
+    openupgrade.logged_query(
+        env.cr,
+        """
+        CREATE TABLE IF NOT EXISTS product_template_sale_product_company_company_rel
```

Comment:
Wait, you should simply rename the table, not create + insert.

Reply:
In any case data will be wrong, so better to raise an exception doing the update.

Reply:
Rename table done